dali/pipeline/operator/sequence_shape_test.cc

```diff
@@ -135,155 +150,289 @@ void check_batch_props(ContainerT &batch, ContainerT &unfolded_batch, int num_sl
       size_t stride = type_size * volume(slice_shapes[sample_idx]);
-      for (int i = 0; i < volume(groups[sample_idx]); i++) {
-        auto ptr = static_cast<uint8_t *>(unfolded_batch[slice_idx++].raw_mutable_data());
+      for (int i = 0; i < volume(unfolded_extents[sample_idx]); i++) {
+        auto ptr = static_cast<uint8_t *>(unfolded_batch.raw_mutable_tensor(slice_idx));
         EXPECT_EQ(ptr, base_ptr + stride * i);
+        EXPECT_EQ(slice_shapes[sample_idx], unfolded_batch.tensor_shape(slice_idx++));
       }
     }
   }
+
+  template <typename Backend>
+  void UnfoldExtents(TensorVector<Backend> &unfolded_batch, const TensorVector<Backend> &batch,
+                     TensorListShape<> unfolded_extents) {
+    unfold_outer_dims(unfolded_batch, batch, unfolded_extents.sample_dim(),
+                      unfolded_extents.num_elements());
+  }
+
+  template <typename Backend>
+  void UnfoldExtents(TensorList<Backend> &unfolded_batch, const TensorList<Backend> &batch,
+                     TensorListShape<> unfolded_extents) {
+    unfold_outer_dims(unfolded_batch, batch, unfolded_extents.sample_dim());
+  }
 };
 
-using Backends = ::testing::Types<CPUBackend, GPUBackend>;
+using Containers = ::testing::Types<TensorVector<CPUBackend>, TensorVector<GPUBackend>,
+                                    TensorList<CPUBackend>, TensorList<GPUBackend>>;
+
+TYPED_TEST_SUITE(SequenceShapeUnfoldTest, Containers);
 
-TYPED_TEST_SUITE(SequenceShapeUnfoldTVTest, Backends);
+TYPED_TEST(SequenceShapeUnfoldTest, Unfold0Extents) {
+  auto [batch, expanded_batch] = this->CreateTestBatch(DALI_UINT32);
+  this->TestUnfolding(*expanded_batch, batch, 0);
+}
+
+TYPED_TEST(SequenceShapeUnfoldTest, Unfold1Extent) {
+  auto [batch, expanded_batch] = this->CreateTestBatch(DALI_UINT16);
+  this->TestUnfolding(*expanded_batch, batch, 1);
+}
 
-TYPED_TEST(SequenceShapeUnfoldTVTest, Unfold0Extents) {
-  auto batch = this->CreateTestBatch(DALI_UINT32);
-  this->TestUnfolding(batch, 0);
+TYPED_TEST(SequenceShapeUnfoldTest, Unfold2Extents) {
+  auto [batch, expanded_batch] = this->CreateTestBatch(DALI_UINT8);
+  this->TestUnfolding(*expanded_batch, batch, 2);
 }
 
-TYPED_TEST(SequenceShapeUnfoldTVTest, Unfold1Extent) {
-  auto batch = this->CreateTestBatch(DALI_UINT16);
-  this->TestUnfolding(batch, 1);
+TYPED_TEST(SequenceShapeUnfoldTest, Unfold2ExtentsPinned) {
+  auto [batch, expanded_batch] = this->CreateTestBatch(DALI_UINT8, true);
+  this->TestUnfolding(*expanded_batch, batch, 2);
 }
 
-TYPED_TEST(SequenceShapeUnfoldTVTest, Unfold2Extents) {
-  auto batch = this->CreateTestBatch(DALI_UINT8);
-  this->TestUnfolding(batch, 2);
+TYPED_TEST(SequenceShapeUnfoldTest, Unfold2Extents3Iters) {
+  auto [batch, expanded_batch] = this->CreateTestBatch(DALI_UINT8, false, "XYZ");
+  this->TestUnfolding(*expanded_batch, batch, 2);
+  batch.Resize({{2, 2, 6}, {13, 4, 11}}, DALI_FLOAT);
+  batch.SetLayout("XYZ");
+  this->TestUnfolding(*expanded_batch, batch, 2);
```

Comment:
```suggestion
  this->TestUnfolding(*expanded_batch, batch, 1);
  this->TestUnfolding(*expanded_batch, batch, 2);
```
Can you also check unfolding of just one dim? I think these tests are cheap and unfolding 1 dim is where we want the best possible coverage.

Reply:
There is a separate test case that just unfolds 1 dim (and the same for all three). I added two cases with multiple iters that test 1, 2 and 3 unfolds in a single test case between resizes.